### PR TITLE
Fixing and adding notes to the recursive method

### PR DESCRIPTION
**Pull request by one-third-reason**
In the examples of the recursive method output, the function was written as "fib_iterative()" instead of "fib_recursive()"
The recursion depth limit thing is added too as a note, and a way to increase the limit is also added

Also I'm not a native english speaker so there might be some wrong gramars but I hope there are none

### Describe your change:

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/fibonacci.py [hunk 1]
```diff
@@ -80,6 +80,14 @@ def fib_recursive(n: int) -> list[int]:
...
         """
         Calculates the i-th (0-indexed) Fibonacci number using recursion
+        
+        NOTE: The current recursion depth limit is 1000 (thus n > 1000 won't work) and it will throw an error if it's exceeded to prevent infinite recursion / stack overflow.
+              The limit can be changed with the sys module, but setting it too high might evoke a crash:
+          
+              import sys
+              sys.setrecursionlimit(limit)
+          
+              A better solution is to rewrite the code without recursion e.g. with loops, like the iterative method.
```
Comment (pandrey2003) [inline maths/fibonacci.py:90]: Two points:

1. may not be a good idea to exceed 100 symbols/line.
2. won't this chunk raise a `sphinx` warning if we try generating docs?

Comment (XyeetD) [reply maths/fibonacci.py:90]: I'm sorry but may you explain what a sphinx warning is? Thank you

Comment (pandrey2003) [reply maths/fibonacci.py:90]: Documentation generator ([link](https://www.sphinx-doc.org/en/master/usage/quickstart.html)).

OK, we can skip the 2nd point, just ensure each line is not longer than 100 symbols

Comment (XyeetD) [reply maths/fibonacci.py:90]: Ah ok I'll try shorten those texts

Comment (pandrey2003) [reply maths/fibonacci.py:90]: Or try moving text to the next line

Comment (XyeetD) [reply maths/fibonacci.py]: > Or try moving text to the next line

Good idea 😅

Comment (pandrey2003) [reply maths/fibonacci.py]: Now, it does not exceed. I hope docs are only for developers to see, so now it's pretty fine regarding length

**Review by pandrey2003 — APPROVED** [review]
Looks pretty fine, at least for a stranger like me :)

**Comment by pandrey2003** [comment]
Need to get a review of someone more knowledgeable, like @Kush1101

**Comment by cclauss** [comment]
In preparation for that denial-of-service attack that is [Hacktoberfest](https://hacktoberfest.com), I am closing all pull requests that do not have passing tests.

**Pull request closed without merging**